testing/unittests/metrics/MetricFrameworkTests.cpp

```diff
@@ -308,6 +309,65 @@ class MetricFrameworkTests : public CppUnit::TestFixture
         CPPUNIT_ASSERT_EQUAL(0, static_cast<int>(pCounterGauge->queryValue()));
     }
 
+
+    void Test_histogram_metric()
+    {
+        std::vector<BucketDef> bucketDefs = {{"le 2", 2}, {"le 4", 4}, {"le 8", 8} };
+        std::shared_ptr<HistogramMetric> pHistogram = std::make_shared<HistogramMetric>("requests.dist", "description", SMeasureCount, bucketDefs);
+
+        std::vector<__uint64> values;
+
+        //
+        // Verify that the correct number of buckets is returned
+        values = pHistogram->queryHistogramValues();
+        CPPUNIT_ASSERT_EQUAL(bucketDefs.size()+1, values.size());
+
+        //
+        // Add a measurement and make sure it goes into the correct bucket
+        int sumMeasurements = 0;
+        int countMeasurements = 0;
+
+        pHistogram->recordMeasurement(4);
+        sumMeasurements += 4;
+        countMeasurements++;
+        checkHistogramBucketResult(pHistogram, sumMeasurements, countMeasurements, {0,1,0,0});
+
+        //
+        // Add same measurement and make sure the bucket increments
+        pHistogram->recordMeasurement(4);
+        sumMeasurements += 4;
+        countMeasurements++;
+        checkHistogramBucketResult(pHistogram, sumMeasurements, countMeasurements, {0,2,0,0});
+
+        //
+        // Add a measurement that is beyond the highest bucket and make sure it goes in the
+        // last "inf" bucket
+        pHistogram->recordMeasurement(20);
+        sumMeasurements += 20;
+        countMeasurements++;
+        checkHistogramBucketResult(pHistogram, sumMeasurements, countMeasurements, {0,2,0,1});
+
+        //
+        // Just to make sure, increment another bucket
+        pHistogram->recordMeasurement(8);
+        sumMeasurements += 8;
+        countMeasurements++;
+        checkHistogramBucketResult(pHistogram, sumMeasurements, countMeasurements, {0,2,1,1});
+    }
+
+
+    void checkHistogramBucketResult(std::shared_ptr<HistogramMetric> pHistogram, int expectedSum, int expectedCount, const std::vector<int> &expectedValues)
```

Comment:
This function needs to take an IMetric as a parameter - so it can match the logic that the sinks need to execute.  (The functions are not currently accessible to the sinks...)

Reply:
Converted parameter to IMetric as requested. Add dynamic pointer cast to histogram metric type to gain access to histogram specific methods. 

It is expected that a sink will check the type of each metric in it's report list and for any encountered histograms, execute a dynamic pointer cast to convert the pointer type and gain access to the necessary methods.